### PR TITLE
chore: bump k8s dependencies to v1.30 and VK to v1.11.0

### PR DESCRIPTION
**Pull request by marwanad**
This PR bumps k8s dependencies to 1.30 and updates the test matrix. 

It also updates the VK version to fix an issue with port-forwarding due to unsupported SPDY protocols.

### CODE REVIEW

**Comment by marwanad** [comment]
Note: Later controller runtime version doesn't play with VK because of some changed method signatures - I'm gonna address this in a follow-up.

File: hack/boilerplate.go.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020 The Multicluster-Scheduler Authors.
+ * Copyright The Multicluster-Scheduler Authors.
```
Comment (marwanad) [inline hack/boilerplate.go.txt:2]: I dropped the year from the boilerplate header.

File: hack/update-codegen.sh [hunk 2]
```diff
@@ -25,17 +25,19 @@ GOPATH="${GOPATH:-"$HOME/go"}"
...
   cd "${SCRIPT_ROOT}"
-  ls -d -1 $GOPATH/pkg/mod/k8s.io/code-generator@v0.27.4 2>/dev/null || echo ../code-generator
+  ls -d -1 $GOPATH/pkg/mod/k8s.io/code-generator@v0.30.5 2>/dev/null || echo ../code-generator
 )}
 
-# generate the code with:
-# --output-base    because this script should also be able to run inside the vendor dir of
-#                  k8s.io/kubernetes. The output-base is needed for the generators to output into the vendor dir
-#                  instead of the $GOPATH directly. For normal projects this can be dropped.
-GOPATH=$GOPATH bash "${CODEGEN_PKG}"/generate-groups.sh "deepcopy,client,informer,lister" \
```
Comment (marwanad) [inline hack/update-codegen.sh:35]: these are all deprecated and removed.

File: test/e2e/kind.sh [hunk 3]
```diff
@@ -19,20 +19,17 @@ set -euo pipefail
...
-# images built for kind v0.20.0
-# https://github.com/kubernetes-sigs/kind/releases/tag/v0.20.0
+# images built for kind v0.24.0
+# https://github.com/kubernetes-sigs/kind/releases/tag/v0.24.0
 declare -A kind_images
 
-kind_images[1.28]="kindest/node:v1.28.0@sha256:b7a4cad12c197af3ba43202d3efe03246b3f0793f162afb40a33c923952d5b31"
+kind_images[1.30]="kindest/node:v1.30.4@sha256:976ea815844d5fa93be213437e3ff5754cd599b040946b5cca43ca45c2047114"
+kind_images[1.29]="kindest/node:v1.29.8@sha256:d46b7aa29567e93b27f7531d258c372e829d7224b25e3fc6ffdefed12476d3aa"
+kind_images[1.28]="kindest/node:v1.28.13@sha256:45d319897776e11167e4698f6b14938eb4d52eb381d9e3d7a9086c16c69a8110"
```
Comment (marwanad) [inline test/e2e/kind.sh:28]: im removing versions below 1.27 from the test matrix

File: test/e2e/kind.sh [hunk 4]
```diff
@@ -19,20 +19,17 @@ set -euo pipefail
...
-kind_images[1.22]="kindest/node:v1.22.17@sha256:f5b2e5698c6c9d6d0adc419c0deae21a425c07d81bbf3b6a6834042f25d4fba2"
-kind_images[1.21]="kindest/node:v1.21.14@sha256:8a4e9bb3f415d2bb81629ce33ef9c76ba514c14d707f9797a01e3216376ba093"
+kind_images[1.30]="kindest/node:v1.30.4@sha256:976ea815844d5fa93be213437e3ff5754cd599b040946b5cca43ca45c2047114"
+kind_images[1.29]="kindest/node:v1.29.8@sha256:d46b7aa29567e93b27f7531d258c372e829d7224b25e3fc6ffdefed12476d3aa"
+kind_images[1.28]="kindest/node:v1.28.13@sha256:45d319897776e11167e4698f6b14938eb4d52eb381d9e3d7a9086c16c69a8110"
+kind_images[1.27]="kindest/node:v1.27.16@sha256:3fd82731af34efe19cd54ea5c25e882985bafa2c9baefe14f8deab1737d9fabe"
 
-K8S_VERSION="${K8S_VERSION:-"1.28"}"
+
+K8S_VERSION="${K8S_VERSION:-"1.29"}"
```
Comment (adrienjt) [inline test/e2e/kind.sh:32]: Why default to version 1.29 and not 1.30?

File: test/e2e/kind.sh [hunk 5]
```diff
@@ -19,20 +19,17 @@ set -euo pipefail
 
 source test/e2e/aliases.sh
 
-# images built for kind v0.20.0
-# https://github.com/kubernetes-sigs/kind/releases/tag/v0.20.0
+# images built for kind v0.24.0
```
Comment (adrienjt) [inline test/e2e/kind.sh:22]: Could you upgrade the versions of kind and kubectl installed in test/e2e/install_dependencies.sh?

**Review by adrienjt — APPROVED** [review]
This is great, thank you so much!

**Comment by adrienjt** [comment]
Have you tried to e2e test with 1.31, even if you can't build against it?

Currently, we're building against 1.27, but testing up to 1.28.

File: test/e2e/kind.sh (hunk 4 above)
Comment (marwanad) [reply test/e2e/kind.sh]: no real reason, just did latest in test matrix - 1. I've switched it to 1.30.

**Comment by marwanad** [comment]
@adrienjt e2e works, I updated this change to support 1.31 in the test matrix.

**Pull request merged**